### PR TITLE
update docs for GetAccessToken() and handle case where auth is nil

### PR DESCRIPTION
**Pull request by k-capehart**

### CODE REVIEW

File: salesforce.go [hunk 1]
```diff
@@ -603,5 +603,8 @@ func (sf *Salesforce) GetJobResults(bulkJobId string) (BulkJobResults, error) {
 }
 
 func (sf *Salesforce) GetAccessToken() string {
-	return sf.auth.AccessToken
+	if sf.auth != nil {
```
Comment (ccoVeille) [inline salesforce.go:606]:
```go
        if sf.auth == nil { 
                 return ""
         } 
         return sf.auth.AccessToken
```

Usually it's written like this for clarity purpose

File: README.md [hunk 2]
```diff
@@ -93,6 +104,16 @@ if err != nil {
 }
 ```
 
+### GetAccessToken()
+
+`func (sf *Salesforce) GetAccessToken() string`
```
Comment (ccoVeille) [inline README.md:109]:
````suggestion
```go
func (sf *Salesforce) GetAccessToken() string
```
````
This way the code will be formatted with Go syntax

File: salesforce.go (hunk 1 above)
Comment (k-capehart) [reply salesforce.go:606]: @ccoVeille Actually, that's a good point. I'll make that update.

File: README.md (hunk 2 above)
Comment (k-capehart) [reply README.md:109]: @ccoVeille I was specifically not formatting it that way for the headers here, but its a good thing to note. Thank you.

**Pull request merged**